src/ast/function/search.rs

```diff
@@ -0,0 +1,31 @@
+use crate::prelude::Column;
+
+#[derive(Debug, Clone, PartialEq)]
+/// Holds the columns on which to perform a full-text search
+pub struct TextSearch<'a> {
+    pub(crate) columns: Vec<Column<'a>>,
+}
+
+/// Performs a full-text search. Use it in combination with the `.matches()` comparable.
+///
+/// ```rust
+/// # use quaint::{ast::*, visitor::{Visitor, Postgres}};
+/// # fn main() -> Result<(), quaint::error::Error> {
+/// let search: Expression = text_search(&vec![Column::from("name"), Column::from("ingredients")]).into();
+/// let query = Select::from_table("recipes").so_that(search.matches("chicken"));
+/// let (sql, params) = Postgres::build(query)?;
+/// assert_eq!("SELECT \"recipes\".* FROM \"recipes\" WHERE to_tsvector(\"name\"|| ' ' ||\"ingredients\") @@ to_tsquery($1)", sql);
+/// assert_eq!(params, vec![Value::from("chicken")]);
+/// # Ok(())    
+/// # }
```

Comment:
I'm actually thinking how the functions here always return a `Function`, but should we actually just always return an `Expression`. Do we ever need the `Function` type returned anywhere?

This would simplify the code a lot:

```rust
let search = text_search(&[Column::from("name"), Column::from("ingredients")]);
let query = Select::from_table("recipes").so_that(search.matches("chicken"));
let (sql, _) = Postgres::build(query)?;

assert_eq!(
    "SELECT \"recipes\".* FROM \"recipes\" WHERE to_tsvector(\"name\"|| ' ' ||\"ingredients\") @@ to_tsquery($1)",
    sql
);
```

See also how the example can be written in a simpler way. No need for `vec!` if you just take a slice, and the pacing and line length should be considered more carefully.